src/MyEventEmitter.js

```diff
@@ -1,12 +1,79 @@
 'use strict';
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  on(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+    this.events[eventName].push(listener);
+  }
+
+  once(eventName, listener) {
+    const onceTimeListener = (...args) => {
+      listener(...args);
+
+      this.off(eventName, onceTimeListener);
+    };
+
+    this.on(eventName, onceTimeListener);
+  }
+
+  off(eventName, listener) {
+    if (!this.events[eventName]) {
+      return;
+    }
+
+    this.events[eventName].filter(listenerName => listenerName !== listener);
+  }
+
+  emit(eventName, ...args) {
+    if (!this.events[eventName]) {
+      return;
+    }
+
+    this.events[eventName].forEach(listener => listener(...args));
+  }
+
+  prependListener(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+
+    this.events[eventName].unshift(listener);
+  }
+
+  prependOnceListener(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+
+    const onceTimeListener = (...args) => {
+      listener(...args);
+
+      this.off(eventName, onceTimeListener);
+    };
+
+    this.prependListener(eventName, onceTimeListener);
+  }
+
+  removeAllListeners(eventName) {
```

Comment:
The else clause in 'removeAllListeners' will clear all listeners for all events, not only for the specified eventName. It would be better to return or throw an error if 'eventName' does not exist in 'this.events'.